### ISSUE TITLE
HMR issues 

### ISSUE THREAD

**Issue by guytepper** [samrum/vite-plugin-web-extension#9]
Hi there!

I ran into some issues with the HMR functionality: 
While watching for file changes and serving the extension on Chrome, the extension will reload itself entirely if a file has been changed, closing the pages I'm working on. 
On Firefox it won't reload the extension entirely, but I will need to manually refresh the page to see the changes.

## How to reproduce
- Create a new project using `npm init @samrum/vite-plugin-web-extension`
- Choose Manifest V2 + V3, Svelte and TypeScript
- Run `npm run watch` and `npm run serve:chrome` parallelly
- Open the extension options page in the browser and open the file `/src/entries/options/App.svelte`
- Make changes in the file - the extension will reload itself entirely
- You can also run `npm run serve:firefox` to see that HMR isn't working on Firefox

Any suggestion how to overcome this?

Thanks a lot for the work on this project! 🙏

**Comment by samrum** [comment]
If you want to use HMR, you need to run in dev mode, not watch mode: `npm run dev`

watch mode will always trigger full extension reloads since it'll rebuild the extension on change.

**Comment by guytepper** [comment]
Thanks! Maybe it's worth adding instructions to the README on how to develop with the plugin?  I was sure `watch` is the script that I should run while I'm working.

Btw, HMR works great on Chrome but Firefox throws this error:
```
Reading manifest: Error processing content_security_policy: ‘script-src’ must include the source 'self'
```
any idea why is that?

**Comment by nilooy** [comment]
> Thanks! Maybe it's worth adding instructions to the README on how to develop with the plugin? I was sure `watch` is the script that I should run while I'm working.
> 
> Btw, HMR works great on Chrome but Firefox throws this error:
> 
> ```
> Reading manifest: Error processing content_security_policy: ‘script-src’ must include the source 'self'
> ```
> 
> any idea why is that?

Same for me as well. I really like the way @samrum has built the plugin but without a fix for the content security, it's unusable.

**Comment by samrum** [comment]
The CSP issue @guytepper mentioned with Firefox is fixed in v0.1.8.
Also, for extensions generated by `init @samrum/vite-plugin-web-extension` there's already a README in the generated extension that explains the different commands and how they work. I might rework them though.

@nilooy's issue was closed with comment: https://github.com/samrum/create-vite-plugin-web-extension/issues/11#issuecomment-1080249439

**Comment by guytepper** [comment]
Thank you so much 🙏💜
